File: templates/hexagonal-spring-boot/rest-adapter/src/main/java/packageName/rest/ExampleResource.java
package packageName.rest;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import packageName.domain.model.ExampleInfo;
import packageName.domain.port.RequestExample;

@RestController
@RequestMapping("/api/v1/examples")
public class ExampleResource {

  private RequestExample requestExample;

  public ExampleResource(RequestExample requestExample) {
    this.requestExample = requestExample;
  }

  @GetMapping
  public ResponseEntity<ExampleInfo> getExamples() {
    return ResponseEntity.ok(requestExample.getExamples());
  }
}

File: templates/hexagonal-spring-boot-kotlin/rest-adapter/src/main/kotlin/packageName/rest/ExampleResource.kt
package packageName.rest

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import packageName.domain.model.ExampleInfo
import packageName.domain.port.RequestExample

@RestController
@RequestMapping("/api/v1/examples")
class ExampleResource(private val requestExample: RequestExample) {

    @GetMapping
    fun getExamples(): ResponseEntity<ExampleInfo> {
        return ResponseEntity.ok(requestExample.getExamples())
    }
}
